### PR TITLE
Capitalize headings rule

### LINKED ISSUE #8: FR: caltialize headers rule
Nice plug-in!

How difficult is it to add new rules? One that will reformat headers to Capitalized would be great!

### PR DESCRIPTION
**Pull request by gavinmn**
Closes #8 

This PR adds the `Capitalize headings` rule with options to format headings where the `First word is capitalized`, in `Proper Title Capitalization`, or in `ALL CAPS`.

### CODE REVIEW

File: rules.ts [hunk 1]
```diff
@@ -279,6 +279,79 @@ export const rules: Rule[] = [
...
+					lines[i] = lines[i].replace(/^#*\s\w.*/, string => string.toUpperCase()) // convert full heading to uppercase
+				} else {
+					lines[i] = lines[i].replace(/^#*\s([a-z])/, string => string.toUpperCase()) // capitalize first letter of heading
+				}
+			}
+			return lines.join("\n");
+		},
+		[
+			new Example(
+				"Headings should be surrounded by blank lines",
```
Comment (platers) [inline rules.ts:316]: Update example description

File: rules.ts [hunk 2]
```diff
@@ -279,6 +279,79 @@ export const rules: Rule[] = [
...
+						}
+						headerWords[j] = headerWords[j].replace(/^\w/, c => c.toUpperCase());
+					}
+
+					lines[i] = lines[i].replace(headerRegex, `$1${headerWords.join(" ")}`);
+
+				} else if (options["allCaps"] == "true") {
+					lines[i] = lines[i].replace(/^#*\s\w.*/, string => string.toUpperCase()) // convert full heading to uppercase
+				} else {
+					lines[i] = lines[i].replace(/^#*\s([a-z])/, string => string.toUpperCase()) // capitalize first letter of heading
```
Comment (platers) [inline rules.ts:309]: I think first letter should always be capital? For headings that start with an ignored word.

File: rules.ts [hunk 3]
```diff
@@ -279,6 +279,79 @@ export const rules: Rule[] = [
...
+				`,
+				dedent`
+				# THIS IS A HEADING 1
+				## THIS IS A HEADING 2
+				`,
+				{allCaps: "true"}
+			),
+		],
+		[
+			"titleCase: Format headings with title case capitalization, default=`false`", "allCaps: Format headings with all capitals, default= `false`"
```
Comment (platers) [inline rules.ts:352]: just for readability, could you add a newline between the options? thanks :)

**Review by platers — COMMENTED** [review]
Thanks! Looks good, just a few comments.

**Comment by gavinmn** [comment]
> Thanks! Looks good, just a few comments.

Updated to address your comments!

**Pull request merged**